### PR TITLE
[MOB-17170] Documentation getting-started and api-reference

### PR DESCRIPTION
**Pull request by kevinlind**
## Description

Adds `Documentation` folder with `api-reference.md` and `getting-started.md` docs.
The majority of the documentation was taken from the public Git Book site. The API references for `getLocationHint` and `setLocationHint` are new.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: Documentation/getting-started.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
+# Getting started
+
+## Before starting
+
+The Adobe Experience Platform Edge Network extension has the following peer dependencies, which must be installed to use the Edge Network extension:
+- [AEPCore](https://github.com/adobe/aepsdk-core-ios#readme)
+- [AEPEdgeIdentity](https://github.com/adobe/aepsdk-edgeidentity-ios#readme)
+
+## Configure the Edge Network extension in Data Collection UI
+1. Log into [Adobe Experience Cloud](https://experience.adobe.com) and select **Data Collection** under **Quick Access**.
```
Comment (cacheung) [inline Documentation/getting-started.md:10]: Quick access (to match the one on https://experience.adobe.com)

File: Documentation/getting-started.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+### Domain configuration
+
+The value under the **Edge Network domain** field is used for requests to Adobe Experience Platform Edge Network and it usually follows the format `<company>.data.adobedc.net`, where `<company>` is the unique namespace associated to your Adobe organization.
+
+If you have a first-party domain mapped to the Adobe-provisioned Edge Network domain, you can enter it here. For more details about how to configure or maintain a first-party domain, see [Adobe-Managed Certificate Program](https://experienceleague.adobe.com/docs/core-services/interface/administration/ec-cookies/cookies-first-party.html?lang=en#adobe-managed-certificate-program).
+
+> **Note**
+> The domain name is expected to be just the domain without any protocol or trailing slashes. If no domain is provided, by default the `edge.adobedc.net` domain is used.
+
+## Add the AEP Edge extension to your app
```
Comment (cacheung) [inline Documentation/getting-started.md:40]: Why we don't use "Add the Edge Network extension to your app" as gitbook?

File: Documentation/getting-started.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+If no datastream was previously created, see [Configure datastreams](https://aep-sdks.gitbook.io/docs/getting-started/configure-datastreams) before moving to the next step.
+
+Select the desired datastream per each environment from the corresponding drop-down lists. If your organization uses multiple sandboxes, select the **Sandbox** first, then select the **Datastream** for each environment.
+
+The datastream used by the client-side implementation is one of the following:
+
+- the _Production_ environment configuration when the Data Collection Tags library is published to production (in the Published column in the publishing flow).
+- the _Staging_ environment configuration when the Data Collection Tags library is published to staging (in the Submitted column in the publishing flow).
+- the _Development_ environment configuration when the Data Collection Tags library is in development.
```
Comment (cacheung) [inline Documentation/getting-started.md:29]: the _Development_ environment configuration when the Data Collection Tags library is in development (in the Development column in the publishing flow).

File: Documentation/api-reference.md [hunk 4]
```diff
@@ -0,0 +1,377 @@
...
+                               _ completion: (() -> Void)? = nil)
+```
+
+##### Example
+```swift
+import AEPEdge
+
+...
+MobileCore.registerExtensions([Edge.self])
+
```
Comment (cacheung) [inline Documentation/api-reference.md:111]: Missing close code view '''

File: Documentation/api-reference.md [hunk 5]
```diff
@@ -0,0 +1,377 @@
...
+##### Syntax
```
Comment (cacheung) [inline Documentation/api-reference.md:112]: Missing
#### Objective-C

Before Syntax

File: Documentation/api-reference.md [hunk 6]
```diff
@@ -0,0 +1,377 @@
...
+```objectivec
++ (void) registerExtensions: (NSArray<Class*>* _Nonnull) extensions
+                 completion: (void (^ _Nullable)(void)) completion;
+
```
Comment (cacheung) [inline Documentation/api-reference.md:116]: missing '''.

File: Documentation/api-reference.md [hunk 7]
```diff
@@ -0,0 +1,377 @@
...
+    /// This is a system generated identifier for the Dataset the event belongs to.
+    /// - Returns: The  identifier as a String for this dataset
+    var datasetIdentifier: String { get }
+}
+```
+
+### EdgeEventHandle
+
+The `EdgeEventHandle` is a response fragment from Adobe Experience Platform Edge Network for a sent XDM Experience Event. One event can receive none, one or multiple `EdgeEventHandle`(s) as response.
+Use this class when calling the[sendEvent](#sendevent) API with `EdgeCallback`.
```
Comment (cacheung) [inline Documentation/api-reference.md:256]: add a space between the [sendEvent]

File: Documentation/api-reference.md [hunk 8]
```diff
@@ -0,0 +1,380 @@
...
+// example 2 - send the experience event and handle the Edge Network response onComplete
+[AEPMobileEdge sendExperienceEvent:event completion:^(NSArray<AEPEdgeEventHandle *> * _Nonnull handles) {
+  // handle the Edge Network response
+}];
+```
+------
+
+### setLocationHint
+
+Set the Edge Network location hint used in requests to the Adobe Experience Platform Edge Network. Passing nil or an empty string will clear the existing location hint. Edge Network responses may overwrite the location hint to a new value when necessary to manage network traffic.
```
Comment (emdobrin) [inline Documentation/api-reference.md:197]: - Set -> Sets the ..
- Passing nil or an empty string will clear the ... -> clears the existing..

File: Documentation/getting-started.md [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+The value under the **Edge Network domain** field is used for requests to Adobe Experience Platform Edge Network and it usually follows the format `<company>.data.adobedc.net`, where `<company>` is the unique namespace associated to your Adobe organization.
+
+If you have a first-party domain mapped to the Adobe-provisioned Edge Network domain, you can enter it here. For more details about how to configure or maintain a first-party domain, see [Adobe-Managed Certificate Program](https://experienceleague.adobe.com/docs/core-services/interface/administration/ec-cookies/cookies-first-party.html?lang=en#adobe-managed-certificate-program).
+
+> **Note**
+> The domain name is expected to be just the domain without any protocol or trailing slashes. If no domain is provided, by default the `edge.adobedc.net` domain is used.
+
+## Add the Edge Network extension to your app
+
+### Download and import the Edge extension
```
Comment (emdobrin) [inline Documentation/getting-started.md:42]: Can we also add SPM instructions first, followed by CocoaPods?

File: Documentation/getting-started.md [hunk 10]
```diff
@@ -0,0 +1,100 @@
+# Getting started
```
Comment (emdobrin) [inline Documentation/getting-started.md:1]: Let's add a README.md file with table of content at Documentation top level. Might want to merge dev changes first.

File: Documentation/getting-started.md [hunk 11]
```diff
@@ -39,9 +39,35 @@ If you have a first-party domain mapped to the Adobe-provisioned Edge Network do
...
+
+Enter the URL for the AEPEdge package repository: `https://github.com/adobe/aepsdk-edge-ios.git`.
+
+When prompted, input a specific version or a range of versions for Version rule.
+
+Alternatively, if your project has a `Package.swift` file, you can add AEPEdge directly to your dependencies:
+
+```swift
+dependencies: [
+  .package(url: "https://github.com/adobe/aepsdk-edge-ios.git", .upToNextMajor(from: "1.1.0"))
```
Comment (emdobrin) [inline Documentation/getting-started.md:61]: we should include core and edgeidentity too

**Review by emdobrin — APPROVED** [review]

**Pull request merged**